src/transaction_builder.js

```diff
@@ -153,33 +172,252 @@ export class TransactionBuilder {
    * @see TimeoutInfinite
    */
   setTimeout(timeout) {
+    if (timeout < 0) {
+      throw new Error('timeout cannot be negative');
+    }
+
+    return this.setMaxTime(Math.floor(Date.now() / 1000) + timeout);
+  }
+
+  /**
+   * If you want to prepare a transaction which will become valid at some point in the future, you can set a minimum time precondition.
+   * Internally this will set the <code>minTime</code> precondition.
+   *
+   * @param {Date|number} dateOrEpochSeconds Either a js Date object, or a number of unix epoch seconds. Can't be negative.
+   * If the value is `0`, the transaction is valid immediately.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinTime(dateOrEpochSeconds) {
+    // Force it to a date type
+    if (typeof dateOrEpochSeconds === 'number') {
+      dateOrEpochSeconds = new Date(dateOrEpochSeconds * 1000);
+    }
+
+    // Convert that date to the epoch seconds
+    const t = Math.floor(dateOrEpochSeconds / 1000);
```

Comment:
Erm is `Date` divided by `int` well-defined? :detective:

Reply:
afaik yes. but added a `.valueOf()` to be more explicit.